### ISSUE TITLE
On click modal

### ISSUE THREAD

**Issue by malinged** [rcaferati/react-awesome-slider#48]
Hello I am trying to get a different modal element to go off OnClick for each image.

```
<AwesomeSlider onClick={this.openModal} >
<div data-src={serene} />
<div data-src={bw1} />
<div data-src={fexivus} />
<div data-src={mount} />
<div data-src={orbfall} />
</AwesomeSlider>
```
OR
```
<AwesomeSlider >
 <div onClick={this.openModal} data-src={serene} />
<div data-src={bw1} />
<div data-src={fexivus} />
<div data-src={mount} />
<div data-src={orbfall} />
</AwesomeSlider>
```

Does not work.

How do I attach event handling to each image in the slider. 

Great app BTW <3

Sorry if this is an ignorant question.

**Comment by Edeph** [comment]
I also want a custom onClick event to trigger stuff based on what image is displayed and I'm not sure how to do that.

**Comment by rcaferati** [comment]
Hey @malinged, thanks for raising it.

On the version that you're using, you could add a child to the modal component with an onClick attached to it, but that wouldn't be optimal.

I've added a fix on `v2.1.1` and now the onClick prop on the root children will behave as expected.
